bigquery_etl/util/common.py

```diff
@@ -162,6 +163,120 @@ def write_sql(output_dir, full_table_id, basename, sql, skip_existing=False):
         f.write("\n")
 
 
+def qualify_table_references_in_path(path: Path) -> str:
+    """Add project id and dataset id to table/view references and persistent udfs in a given query.
+
+    e.g.:
+    `table` -> `target_project.default_dataset.table`
+    `dataset.table` -> `target_project.dataset.table`
+
+    This allows a query to run in a different project than the sql dir it is located in
+    while referencing the same tables.
+    """
+    # sqlglot cannot handle scripts with variables and control statements
+    if re.search(
+        r"^\s*DECLARE\b", path.read_text(), flags=re.MULTILINE
+    ) or path.name in ("script.sql", "udf.sql"):
+        raise NotImplementedError(
+            "Cannot qualify table_references of query scripts or UDFs"
+        )
+
+    # determine the default target project and dataset from the path
+    target_project = Path(path).parent.parent.parent.name
+    default_dataset = Path(path).parent.parent.name
+
+    # sqlglot doesn't support Jinja, so we need to render the queries and
+    # init queries to raw SQL
+    sql_query = render(
+        path.name,
+        template_folder=path.parent,
+        format=False,
+        **DEFAULT_QUERY_TEMPLATE_VARS,
+    )
+    init_query = render(
+        path.name,
+        template_folder=path.parent,
+        format=False,
+        **{
+            "is_init": lambda: True,
+            "metrics": MetricHub(),
+        },
```

Comment:
nit: why not
```suggestion
       is_init=lambda: True,
       metrics=MetricHub(),
```

Reply:
counterpoint: why not
```suggestion
        *list((path.name,)),
        **{
            "template_folder": path.parent,
            "format": False,
            "is_init": lambda: True,
            "metrics": MetricHub(),
        },
```

Reply:
Very nit, but passing in a dict literal with `**{...}` seems strange. Above `**DEFAULT_QUERY_TEMPLATE_VARS` makes sense to me because it's a constant (a variable would also make sense), but for a literal, I can't see how it's better than just named args